app/views/play/level/LevelGoalsView.coffee

```diff
@@ -53,7 +53,7 @@ module.exports = class LevelGoalsView extends CocoView
     @previousGoalStatus ?= {}
     @succeeded = e.overallStatus is 'success'
     for goal in e.goals
-      state = e.goalStates[goal.id]
+      state = e.goalStates[goal.id] ? { status: 'incomplete' }
```

Comment:
Why is this change required?

Reply:
I will review this again after more modifications to the level loading and goal manager were done. Previously, there would be scenarios where goals were not yet loaded, and it would fail here.

Reply:
We want to make sure that this does not affect the existing levels